source/common/stats/stats_impl.h

```diff
@@ -60,12 +64,27 @@ class TagProducerImpl : public TagProducer {
    */
   std::string produceTags(const std::string& metric_name, std::vector<Tag>& tags) const override;
 
+  void addExtractor(TagExtractorPtr extractor);
+  void addExtractorsMatching(absl::string_view name);
```

Comment:
Please comment up all new functions.

Existing Envoy code has a pretty low bar for comments but I'd like to raise it :-)
It also makes things easier on your reviewers - I'll do one more pass with comments added as it'll be faster than you waiting on me reading and grokking all of the tag code :-)